lib/PicoDb/StatementHandler.php

```diff
@@ -285,7 +296,13 @@ protected function bindParams(PDOStatement $pdoStatement)
     protected function beforeExecute()
     {
         if ($this->logQueries) {
-            $this->db->setLogMessage($this->sql);
+            $sql = $this->sql;
+            if ($this->logQueryValues) {
+                foreach ($this->lobParams as $value) {
+                    $sql = substr_replace($sql, "'$value'", strpos($sql, '?'), 1);
```

Comment:
This will work in most cases, but what if the replacement contains `?`. I'd imagine next replacement will be injected into it.

Could this be refactored to detect all `?` characters first, and then for each replace them with the `lobParams` at the matching index? Using `preg_replace_callback` might simplify things.

Reply:
yeah thats probably a good idea. FYI I copied this code from somewhere else in this repo used with explain, wich would have the same issue.